pkg/zql/processors/README.md

```diff
@@ -0,0 +1,296 @@
+# Processors
+
+A pipeline may contain one or more _processors_ to filter or transform event data. You can imagine the data flowing left-to-right through a processor, with its functionality further determined by arguments you may set. Processor names are case-insensitive.
+
+The following available processors are documented in detail below:
+
+* [`cut`](#cut)
+* [`filter`](#filter)
+* [`head`](#head)
+* [`sort`](#sort)
+* [`tail`](#tail)
+* [`uniq`](#uniq)
+
+**Note**: In the examples below, we'll use the `zq -f table` output format for human readability. Due to the width of the Zeek events used as sample data, you may need to "scroll right" in the output to see some field values.
+
+---
+
+## `cut`
+
+|                           |                                                             |
+| ------------------------- | ----------------------------------------------------------- |
+| **Description**           | Return the data only from the specified named fields.       |
+| **Syntax**                | `cut <field-list>`                                          |
+| **Required<br>arguments** | `<field-list>`<br>One or more comma-separated field names.  |
+| **Optional<br>arguments** | None                                                        |
+| **Caveats**               | The specified field names must exist in the input data. If a non-existent field appears in the `<field-list>`, the returned results will be empty. |
+| **Developer Docs**        | https://godoc.org/github.com/mccanne/zq/proc#Cut            |
+
+#### Example:
+
+To return only the `ts` and `uid` columns of `conn` events:
+
+```
+zq -f table '* | cut ts,uid' conn.log
+```
+
+#### Output:
+```
+TS                UID
+1521911721.255387 C8Tful1TvM3Zf5x8fl
+1521911721.411148 CXWfTK3LRdiuQxBbM6
+1521911721.926018 CM59GGQhNEoKONb5i
+...
+```
+
+---
+
+## `filter`
+
+|                           |                                                                       |
+| ------------------------- | --------------------------------------------------------------------- |
+| **Description**           | Apply a search expression to potentially trim data from the pipeline. |
+| **Syntax**                | `filter <search-expression>`                                          |
+| **Required<br>arguments** | `<search-expression>`<br>Any valid expression in ZQL [search syntax](../search-syntax/README.md) |
+| **Optional<br>arguments** | None                                                                  |
+| **Developer Docs**        | https://godoc.org/github.com/mccanne/zq/proc#Filter                   |
+
+#### Example #1:
+
+To further trim the data returned in our [`cut`](#cut) example:
+
+```
+zq -f table '* | cut ts,uid | filter uid=CXWfTK3LRdiuQxBbM6' conn.log
+```
+
+#### Output:
+```
+TS                UID
+1521911721.411148 CXWfTK3LRdiuQxBbM6
+```
+
+#### Example #2:
+
+An alternative syntax for our [`and` operator example](#../search-syntax/README.md#and):
```

Comment:
I started with reading this page so "our and operator example" is unfamiliar to me...

Reply:
Do you see that as a problem? By linking to the Search Syntax page from the table that summarized the proc and also linking to that specific `and` example from it here, I was trying to reinforce the message that there's nothing so special about `filter`, i.e. it certainly can be handy if you have a legit need for downstream additional filtering in a pipeline, but most of the time you probably want to be putting as much filtering logic up front as you can. Now that I'm writing it here, perhaps I should be saying as much in the doc. Let me know if you agree.

Another thing I struggle with sometimes is knowing how much repetition is appropriate, if cross-linking is a good thing or a bad thing, etc. I like the idea of portions of the doc standing alone, but there's also subtle details that are important enough that I'd like to give people a shot of seeing it without having to rewrite a bunch of text and make things bulky (what is a glob wildcard, that the `and` operator can be optionally included, etc.)

Also, since the bullets in the [top-level ZQL readme](https://github.com/mccanne/zq/blob/docs-init/pkg/zql/README.md) link to the Search Syntax page before the Processors page, I guessed I'd hoped that first-time readers might go through them in that order and read each one sequentially, though perhaps that is naive. If I'm wrong about that, it might make an argument for more repetition to make things stand alone and less inter-linking. Hrm.